packages/flutter_web_plugins/lib/src/navigation/utils.dart

```diff
@@ -15,30 +13,30 @@ AnchorElement _urlParsingNode;
 String extractPathname(String url) {
   // TODO(mdebbar): Use the `URI` class instead?
   _urlParsingNode ??= AnchorElement();
-  _urlParsingNode.href = url;
-  final String pathname = _urlParsingNode.pathname;
+  _urlParsingNode!.href = url;
+  final String pathname = _urlParsingNode!.pathname ?? '';
   return (pathname.isEmpty || pathname[0] == '/') ? pathname : '/$pathname';
 }
 
-Element _baseElement;
+Element? _baseElement;
```

Comment:
I'd make this:

```dart
final Element? _baseElement = document.querySelector('base');
```

That would simplify a lot of code below

Reply:
Reading some more code below (`checkBaseHref`), it seems this `base` element needs to be pre-rendered on the index.html, so this should work as expected!